modules/adnowBidAdapter.js

```diff
@@ -0,0 +1,148 @@
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { NATIVE, BANNER } from '../src/mediaTypes.js';
+import * as utils from '../src/utils.js';
+import includes from 'core-js-pure/features/array/includes.js';
+
+const BIDDER_CODE = 'adnow';
+const ENDPOINT = '//n.ads3-adnow.com/a';
+
+/**
+ * @typedef {object} CommonBidData
+ *
+ * @property {string} requestId The specific BidRequest which this bid is aimed at.
+ *   This should match the BidRequest.bidId which this Bid targets.
+ * @property {string} currency The currency code for the cpm value
+ * @property {number} cpm The bid price, in US cents per thousand impressions.
+ * @property {string} creativeId The id of ad content
+ * @property {number} ttl Time-to-live - how long (in seconds) Prebid can use this bid.
+ * @property {boolean} netRevenue Boolean defining whether the bid is Net or Gross.  The default is true (Net).
+ * @property {object} [meta] Object for storing bid meta data
+ * @property {string} [meta.mediaType] banner or native
+ */
+
+/** @type {BidderSpec} */
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [ NATIVE, BANNER ],
+
+  /**
+   * @param {object} bid
+   * @return {boolean}
+   */
+  isBidRequestValid(bid) {
+    if (!bid || !bid.params) return false;
+
+    const codeId = parseInt(bid.params.codeId, 10);
+    if (!codeId) {
+      return false;
+    }
+
+    const mediaType = bid.params.mediaType || NATIVE;
+
+    return includes(this.supportedMediaTypes, mediaType);
+  },
+
+  /**
+   * @param {BidRequest[]} validBidRequests
+   * @param {*} bidderRequest
+   * @return {ServerRequest}
+   */
+  buildRequests(validBidRequests, bidderRequest) {
+    return validBidRequests.map(req => {
+      const mediaType = req.params.mediaType || NATIVE;
```

Comment:
mediaType shouldn't be passed in in your params, it is setup on the adUnit level and passed to the request in bidRequest.mediaTypes. The hello_world.html page returns a native ad in your server response when only sending in a codeId param. Please update to read the mediaTypes object on the bidRequest and remove the 'mediaTypes' param.